src/ConfigParser.h

```diff
@@ -232,7 +232,10 @@ class ConfigParser
     static enum ParsingStates {atParseKey, atParseValue} KvPairState_; ///< Parsing state while parsing kv-pair tokens
```

Comment:
Just to clarify: Current PR description shows the following cache.log messages in the first case/example, but does not show them in later ones:

```
Starting Authentication on port [::]:4443
Disabling Authentication on port [::]:4443 (interception enabled)
```

I left those messages in the first case/example because their presence (or presence of similar messages in different configurations) makes that first problem worse by separating the true problem report from the last log line seen by the admin (where most folks expect to find some explanation for the sudden Squid death).

These messages and other (re)configuration "noise" _are_ present in other examples as well, but I removed them to highlight other specific problems this PR is addressing (and PR description is documenting). This PR does not reduce or add (re)configuration noise.

Reply:
>  This PR does not reduce or add (re)configuration noise.

There are some new messages (error-reporting unrelated) brought about by fatal(), but they are probably expected:
```
storeDirWriteCleanLogs: Starting...
Finished.  Wrote 0 entries.
Took 0.00 seconds (  0.00 entries/sec).
```

Reply:
Yes, and these irrelevant messages are already produced in all other cases when reconfiguration fails (because all other cases end up calling destruct()/fatal()). Since those other cases dominate, I think it is fair to say that this PR does not really add reconfiguration noise.